process/smartContract/hooks/blockChainHook.go

```diff
@@ -467,6 +476,29 @@ func (bh *BlockChainHookImpl) IsPayable(sndAddress []byte, recvAddress []byte) (
 	return metadata.Payable, nil
 }
 
+// FilterCodeMetadataForUpgrade will filter the provided input bytes as a correctly constructed vmcommon.CodeMetadata bytes
+// taking into account the activation flags for the future flags. This should be used in the upgrade SC process
+func (bh *BlockChainHookImpl) FilterCodeMetadataForUpgrade(input []byte) ([]byte, int) {
```

Comment:
I think the question with the upgrade contract with wrong codeMetadata FFFFFF could make a contract non upgradeable 0000?

Reply:
right, this will need to return an error also

Reply:
After `flagFilterCodeMetadataEnableEpoch` becomes true, incorrect code metadata on the `VMOutput` must indeed fail the entire transaction, both for deploy and upgrade.

Reply:
done

Reply:
+1